apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/RichTextEditor/index.tsx

```diff
@@ -0,0 +1,153 @@
+const RichTextEditor = ({
+  csrfToken: _csrfToken,
+  htmlContents,
+}: {
+  htmlContents: string | null;
+  csrfToken: string;
+}) => {
+  const editor = useEditor({
+    editorProps: {
+      attributes: {
+        'aria-label': 'Main content area, start typing to enter text.',
+        autocapitalize: 'off',
+        autocomplete: 'off',
+        autocorrect: 'off',
+      },
+    },
+    emitContentError: true,
+    enableContentCheck: true,
+    immediatelyRender: false,
+    onContentError: (event) => {
+      throw new Error(event.error.message);
+    },
+    onCreate: async ({ editor }) => {
+      const formattedHtml = await formatHtmlWithPrettier(editor.getHTML());
+      setFormattedHtml(formattedHtml);
+    },
+    onUpdate: async ({ editor }) => {
+      const formattedHtml = await formatHtmlWithPrettier(editor.getHTML());
+      setFormattedHtml(formattedHtml);
+    },
+    content: htmlContents,
+    extensions: [
+      ...new Set([
+        Blockquote,
+        Bold,
+        BulletList,
+        Code,
+        Document,
+        Dropcursor,
+        Focus,
+        Gapcursor,
+        Heading,
+        Italic,
+        ListItem,
+        ListKeymap,
+        OrderedList,
+        Paragraph,
+        PlQuestionPanel,
+        Selection,
+        Strike,
+        Subscript,
+        Superscript,
+        Text,
+        Underline,
+        UndoRedo,
+        // For this to work, we need all the nodes to be matched correctly by the other extensions.
+        RawHtml,
+      ]),
+    ],
+  });
+  const [formattedHtml, setFormattedHtml] = useState<string | null>(null);
+  const [debugMode, setDebugMode] = useState<boolean>(false);
+
+  if (htmlContents === null) {
+    return null;
+  }
+
+  if (editor === null) {
+    return null;
+  }
+
+  return (
+    <>
+      <Card class="m-3">
+        <Card.Header>
+          <div class="d-flex align-items-center justify-content-between">
+            Rich Text Editor
+            <Form.Check
+              type="switch"
+              id="rich-text-editor-debug-mode"
+              label="Debug mode"
+              checked={debugMode}
+              onChange={(e) => setDebugMode(e.currentTarget.checked)}
+            />
+          </div>
+        </Card.Header>
+        <Card.Body>
+          <div class="mb-3" />
+          <EditorContent editor={editor} class="border" />
+          {debugMode ? (
+            <Card class="mt-3">
+              <Card.Header>Formatted HTML</Card.Header>
+              <Card.Body>
+                <pre class="mb-0">
+                  <code>{formattedHtml ?? ''}</code>
+                </pre>
+              </Card.Body>
+            </Card>
+          ) : null}
+        </Card.Body>
+        {/* <FloatingMenu editor={editor}>This is the floating menu</FloatingMenu> */}
+        {/* <BubbleMenu editor={editor}>This is the bubble menu</BubbleMenu> */}
+      </Card>
+    </>
+  );
+};
+
+RichTextEditor.displayName = 'RichTextEditor';
+export { RichTextEditor };
```

Comment:
Nit, inline `export const RichTextEditor ...`?

Reply:
Fixed in d4ec582.